igann/igann.py

```diff
@@ -343,7 +343,9 @@ def fit(self, X, y, val_set=None, eval=None, plot_fixed_features=None, fitted_du
         '''
 
         self._reset_state()
-
+        X_copy = deepcopy(X)
```

Comment:
Memory intense, i suspect double the RAM consumption? May cause problems with large datasets, as we saw some datasets already reach 64GB or 128GB fast then.
Maybe there is another solution to converting the datatype or accessing the same variable?

Reply:
Its a point but I do not think its a real concern! Memory consumption of igann is mostly related to the number of regressors  and not the the size of the dataset.  e.g. a 400mb csv with 1million rows and 20float features consumes ~170mb in ram. ref -> ( https://stackoverflow.com/questions/18089667/how-to-estimate-how-much-memory-a-pandas-dataframe-will-need)

Reply:
This only copies the init feature matrix right? I dont think this is too memory intense